### PR TITLE
bazel: add another config_setting incompatible flag

### PR DESCRIPTION
**Pull request by keith**
Turns out the first one I added doesn't apply without this one, which I thought had already been flipped

Signed-off-by: Keith Smiley <keithbsmiley@gmail.com>

### CODE REVIEW

**Comment by keith** [comment]
rules_pkg update blocked on https://github.com/envoyproxy/envoy/pull/23691, trying a patch for now

File: bazel/repositories.bzl [hunk 1]
```diff
@@ -221,7 +221,10 @@ def envoy_dependencies(skip_targets = []):
     external_http_archive("bazel_toolchains")
     external_http_archive("bazel_compdb")
     external_http_archive("envoy_build_tools")
-    external_http_archive("rules_pkg")
+    external_http_archive(
```
Comment (phlax) [inline bazel/repositories.bzl:224]: can we remove the patch once `rules_pkg` has been updated? im thinking we might want to comment that here

Comment (phlax) [reply bazel/repositories.bzl:224]: (i see we can upstream)

**Review by phlax — APPROVED** [review]
lgtm, thanks @keith 

ive left a comment nit - happy to reapprove if you want to add it

**Pull request merged**